community/io/src/main/java/org/neo4j/io/pagecache/tracing/cursor/DefaultPageCursorTracer.java

```
@@ -64,10 +66,12 @@ public void reportEvents()
         if ( hits > 0 )
         {
             pageCacheTracer.hits( hits );
+            historicalHits = Math.addExact( historicalHits, hits );
         }
         if ( faults > 0 )
         {
             pageCacheTracer.faults( faults );
+            historicalFaults = Math.addExact( historicalFaults, faults );
```

Is it better to throw an exception than to overflow in these cases?

---

I don't think signed long overflow is going to be a concern for these cases in practice.

---

yeah exception here is not nice and anyway agree with you that it should not be the case ever anyway